flink-libraries/flink-ml/src/main/scala/org/apache/flink/ml/preprocessing/Splitter.scala

```diff
@@ -0,0 +1,215 @@
+    var dsu = dsa(0)
+    for (k <- 1 to dsa.length-1) {
+      dsu = dsu.union(dsa(k))
+    }
+    dsu
+  }
+
+  // --------------------------------------------------------------------------------------------
+  //  trainTestSplit
+  // --------------------------------------------------------------------------------------------
+  /**
+   * A wrapper for randomSplit that yields a TrainTestDataSet
+   *
+   * @param input           DataSet to be split
+   * @param fraction        Probability that each element is chosen, should be [0,1] without
+   *                        replacement, and [0, ∞) with replacement. While fraction is larger
+   *                        than 1, the elements are expected to be selected multi times into
+   *                        sample on average. This fraction refers to the training element in
+   *                        TrainTestSplit
+   * @param precise         Sampling by default is random and can result in slightly lop-sided
+   *                        sample sets. When precise is true, equal sample set size are forced,
+   *                        however this is somewhat less efficient.
+   * @param seed            Random number generator seed.
+   * @return A TrainTestDataSet
+   */
+  def trainTestSplit[T: TypeInformation : ClassTag]( input: DataSet[T],
+                                                     fraction: Double = 0.6,
+                                                     precise: Boolean = false,
+                                                     seed: Long = Utils.RNG.nextLong())
+  : TrainTestDataSet[T] = {
+    val dataSetArray = randomSplit(input, fraction, precise, seed)
+    TrainTestDataSet(dataSetArray(0), dataSetArray(1))
+  }
+
+  // --------------------------------------------------------------------------------------------
+  //  trainTestHoldoutSplit
+  // --------------------------------------------------------------------------------------------
+  /**
+   * A wrapper for multiRandomSplit that yields a TrainTestHoldoutDataSet
+   *
+   * @param input           DataSet to be split
+   * @param fracArray       An array of length 3, where the first element specifies the size of the
+   *                        training set, the second element the testing set, and the third
+   *                        element is the holdout set. These are proportional and will be
+   *                        normalized internally.
+   * @param precise         Sampling by default is random and can result in slightly lop-sided
+   *                        sample sets. When precise is true, equal sample set size are forced,
+   *                        however this is somewhat less efficient.
+   * @param seed            Random number generator seed.
+   * @return A TrainTestDataSet
+   */
+  def trainTestHoldoutSplit[T: TypeInformation : ClassTag](input: DataSet[T],
+                                                    fracArray: Array[Double] = Array(0.6,0.3,0.1),
```

Comment:
couldn't figure a way to coerce a tuple3 into a typed array that was less invasive than simply throwing the exception.